tests.py

```diff
@@ -1,15 +1,800 @@
+    raise util.APIError(f"Request failed for data type {data_type}!", 400)
+
+
+# TODO: Split tests into different files, one for util,
+# one for CommCareAPIHandler, and one for each
+# lambda function folder (when implemented)
+class TestUtil(unittest.TestCase):
+    def test_process_response(self):
+        print('*** Running test_process_response ***')
+        test_data = [
+            {
+                'name': 'is_boto_200',
+                'parameters': {
+                    'response': {
+                        'ResponseMetadata': {
+                            'HTTPStatusCode': 200
+                        }
+                    },
+                    'is_boto': True
+                },
+                'return_value': None,
+                'expect_exception': False,
+                'exception': None
+            },
+            {
+                'name': 'is_boto_400',
+                'parameters': {
+                    'response': {
+                        'ResponseMetadata': {
+                            'HTTPStatusCode': 400
+                        }
+                    },
+                    'is_boto': True
+                },
+                'return_value': None,
+                'expect_exception': True,
+                'exception': Exception("Boto3 request failed! Code: 400.")
+            },
+            {
+                'name': 'is_not_boto_ok',
+                'parameters': {
+                    'response': MockResponse(),
+                    'is_boto': False
+                },
+                'return_value': {},
+                'expect_exception': False,
+                'exception': None
+            },
+            {
+                'name': 'is_not_boto_not_ok',
+                'parameters': {
+                    'response': MockResponse(
+                        status_code=400, ok=False, reason='Test'),
+                    'is_boto': False
+                },
+                'return_value': None,
+                'expect_exception': True,
+                'exception': util.APIError(
+                    "Request failed! Code: 400. Reason: Test", 400)
+            }
+        ]
+
+        for test_case in test_data:
+            with self.subTest(test_case['name']):
+                print(f'  Running subtest {test_case["name"]}')
+                try:
+                    response = util.process_response(
+                        test_case['parameters']['response'],
+                        test_case['parameters']['is_boto']
+                    )
+                    if test_case['return_value'] is None:
+                        self.assertIsNone(response)
+                    else:
+                        self.assertDictEqual(
+                            test_case['return_value'], response)
+                    self.assertFalse(test_case['expect_exception'])
+                except Exception as e:
+                    self.assertTrue(test_case['expect_exception'])
```

Comment:
I can also probably make a method for handling the tests to reduce the repeated code.  They all follow a super similar structure, so it would be easy enough to do.  I think it makes sense to save that for another tech debt ticket though, so I'll just make a ticket for it for later.

Reply:
I just ended up doing it since it wasn't too much work and made the code much simpler.